### ISSUE TITLE
Calling map/collect with block inside method that accepts block produces unexpected result

### ISSUE THREAD

**Issue by KieranP** [rubocop/rubocop#14835]
## Is your feature request related to a problem? Please describe.

I spent 45 minutes trying to work out why this didn't work:

```ruby
render json: queries.map do |q|
  { _id: q.id }
end
```

Turns out, Ruby is parsing it like this (giving block to the first method called, not the nearest):

```ruby
render(json: queries.map) do |q|
  { _id: q.id }
end
```

And apparently `queries.map` without a block is valid syntax, and returns an array of the raw objects.

Rails then serializes the array of raw objects into a hash containing all the records attributes.

I had to change to this:

```ruby
results = queries.map do |q|
  { _id: q.id }
end

render json: results
```

## Describe the solution you'd like

A Rubocop rule that catches when using a block method (like map, collector, filter, etc) inside another method that also takes a block.

## Describe alternatives you've considered

Is there any? Ruby accepts this syntax, and only becomes evident at runtime.

**Comment by hammadxcm** [comment]
I've opened a PR to address this: #14949

The approach extends the existing `Lint/AmbiguousBlockAssociation` cop to detect `do...end` blocks that are likely intended for an enumerable method in the arguments but actually bind to the outer method call.

For the exact example in this issue:

```ruby
# Flagged — map is called without a block because the do block binds to render
render json: queries.map do |q|
  { _id: q.id }
end
```

The cop will report:

> `map` is called without a block because the `do` block binds to `render`. Use braces or extract to a variable.

It covers `map`, `collect`, `flat_map`, `select`, `filter`, `find`, `each`, `reduce`, `sort_by`, `group_by`, `filter_map`, and other common enumerable methods. It respects the existing `AllowedMethods` and `AllowedPatterns` configuration, and intentionally does not autocorrect since the fix depends on intent.

**Comment by hammadxcm** [comment]
@KieranP  Please confirm if this fixed your issue.
